lib/cognition/plugins/base.rb

```diff
@@ -16,6 +36,36 @@ def self.match(trigger, action, options = {})
       def self.definitions
         @definitions ||= []
       end
+
+      def render(opts = {})
+        options = RENDER_DEFAULTS.merge(opts)
+        calling_method = caller[0][/`([^']*)'/, 1]
+        template = options[:template] || template_file(calling_method, options[:type], options[:engine])
+        @template ||= Tilt.new(template)
```

Comment:
Since `render` takes options, is it safe to memoize `@template`? You'll have the same template object on every call, even if different options are passed.